ERCS/erc-7818.md

```diff
@@ -0,0 +1,162 @@
+     * @notice The transfer "MUST" revert if the token `identifier` is expired.
+     */
+    function transfer(
+        address to,
+        uint256 identifier,
+        uint256 value
+    ) external returns (bool);
+
+    /**
+     * @dev Transfers a specific `identifier` and value from one account to another.
+     * @param from The sender's address.
+     * @param to The recipient's address.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @param value The amount to transfer.
+     * @return bool True if the transfer succeeded, false or reverted if give `identifier` it's expired.
+     * @notice The transfer "MUST" revert if the token `identifier` is expired.
+     */
+    function transferFrom(
+        address from,
+        address to,
+        uint256 identifier,
+        uint256 value
+    ) external returns (bool);
+}
+```
+
+### Behavior specification
+
+- `balanceOf` **MUST** return the total balance of tokens held by an account that are still valid (i.e., have not expired). This includes any tokens associated with specific periods, epochs, or other identifiers, provided they remain within their validity duration. Expired tokens **MUST NOT** be included in the returned balance, ensuring that only actively usable tokens are reflected in the result.
+- `transfer` and `transferFrom` **MUST** exclusively transfer tokens that remain non-expired at the time of the transaction. Attempting to transfer expired tokens **MUST** revert the transaction or return false. Additionally, implementations **MAY** include logic to prioritize the automatic transfer of tokens closest to expiration, ensuring that the earliest expiring tokens are used first, provided they meet the non-expired condition.
+- `totalSupply` **SHOULD** be set to `0` or `type(uint256).max` due to the challenges of tracking only valid (non-expired) tokens.
+- The implementation **MAY** use a standardized revert message, such as `ERC7818TransferredExpiredToken` or `ERC7818TransferredExpiredToken(address sender, uint256 identifier)`, to clearly indicate that the operation failed due to attempting to transfer expired tokens. 
+
+## Rationale
+
+The rationale for developing an expirable [ERC-20](./eip-20.md) token extension is based on several key requirements that ensure its practicality and adaptability for various applications to
+
+### Compatibility with the existing [ERC-20](./erc-20.md) standard. 
+The extension should integrate smoothly with the [ERC-20](./eip-20.md) interface, This ensures compatibility with existing token ecosystems and third-party tools like wallets and blockchain explorers.
+
+### Flexible interface for various implementation. 
+The smart contract should be extensible, allowing businesses to tailor the expiration functionality to their specific needs like expiry in bulk or each token independent expire, whether it’s dynamic reward systems or time-sensitive applications.
+
```

Comment:
You'll definitely want to talk about why you introduced the concept of epochs in here. It doesn't seem necessarily related to expiry.